### PR TITLE
ndcli: allow listing containers in all layer3domains

### PR DESCRIPTION
**Pull request by zeromind**
extend list containers to support listing containers
for all layer3domains

also take user's configured layer3domain into account, via `get_layer3domain()`

always print the layer3domain for `ndcli list containers` to avoid confusion with the output as the user can configure the layer3domain in various places (environment variable, config), that don't show up in the command-line.

only print the layer3domain (name) if it actually has containers

### CODE REVIEW

File: dim-testsuite/t/container-list.t [hunk 1]
```diff
@@ -12,7 +12,25 @@ INFO - Creating container 10.0.0.0/8 in layer3domain one
 $ ndcli create container 11.0.0.0/8 layer3domain one
 INFO - Creating container 11.0.0.0/8 in layer3domain one
 $ ndcli list containers
```
Comment (Srudush) [inline dim-testsuite/t/container-list.t:14]: We should add more containers, pools and delegations here to get all variants of possible output.

```
$ ndcli create container 10.0.4.0/23 layer3domain default
$ ndcli create pool a layer3domain default
$ ndcli modify pool a add subnet 10.0.5.0/24
$ ndcli modify pool a get delegation 27
```

When doing `list containers layer3domain default 10.0.5.0/25` it should then return all found parents and children in that space.

File: dim/dim/rpc.py [hunk 2]
```diff
@@ -3423,6 +3423,17 @@ def has_access(can_create_rr, can_delete_rr):
                     and_(0 < has_access(False, True)).label('can_delete_rr')]
 
 
+    @readonly
+    def ipblock_get_attrs_multi(self, ipblock, layer3domain=None, full=False, filters = {}, **options):
+        logging.warn(filters)
```
Comment (Srudush) [inline dim/dim/rpc.py:3428]: Is this leftover debug output?

File: ndcli/dimcli/__init__.py [hunk 3]
```diff
@@ -1607,12 +1607,20 @@ def print_tree(tree, level):
...
                     print_tree(node['children'], level + 1)
 
-        layer3domains = [args.layer3domain]
-        if args.layer3domain == "all":
-            layer3domains = [l['name'] for l in self.client.layer3domain_list()]
+        if args.layer3domain == "all" or get_layer3domain(args.layer3domain) is None:
+            force_print_layer3domain = True
+            if args.container:
+                filters = dict(status=['Container'])
+                blocks = self.client.ipblock_get_attrs_multi(args.container, filters=filters)
```
Comment (Srudush) [inline ndcli/dimcli/__init__.py:1614]: These two lines can result in objects being missed. The behavior before layer3domain=all was introduced the surrounding containers, subnets and the like were also returned. By searching only for containers, these might get missed and the output wrong.
But that should be found when changing the tests in a way that this situation is provoked.

File: dim/dim/rpc.py (hunk 2 above)
Comment (zeromind) [reply dim/dim/rpc.py]: Yes, I've removed it.

File: ndcli/dimcli/__init__.py (hunk 3 above)
Comment (zeromind) [reply ndcli/dimcli/__init__.py]: Not used anymore, iterating over the layer3domains instead.

File: dim-testsuite/t/container-list.t (hunk 1 above)
Comment (Srudush) [reply dim-testsuite/t/container-list.t:17]: You are right. Thank you :)

**Review by Srudush — CHANGES_REQUESTED** [review]

File: dim/dim/rpc.py [hunk 4]
```diff
@@ -3423,6 +3423,16 @@ def has_access(can_create_rr, can_delete_rr):
                     and_(0 < has_access(False, True)).label('can_delete_rr')]
 
 
+    @readonly
+    def ipblock_get_attrs_multi(self, ipblock, layer3domain=None, full=False, filters = {}, **options):
```
[the remark this answers is not in the document]
Comment (zeromind) [reply dim/dim/rpc.py:3427]: It's not used at the moment, but I figured we might still want to have it for getting results for multiple matching ipblocks.
`ipblock_get_attrs()` does not really work with overlapping across layer3domains.

Can put that into another PR/issue.

**Review by Srudush — APPROVED** [review]

File: dim/dim/rpc.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (Srudush) [reply dim/dim/rpc.py:3427]: Okay, then lets keep it for now.

**Pull request merged**